src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,26 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other != null && getClass() == other.getClass() && this == other) {
```

Comment:
these should be separate checks
they make no sense if put all together

Reply:
done